drain-watch-image/drain-watch.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+CHECK_INTERVAL="${CHECK_INTERVAL:-60}"
+RPC_ADDRESS="${RPC_ADDRESS:-127.0.0.1:24444}"
+
+[ -z "$BUFFER_PATH" ] && exit 2
+
+# this loop will not go on indefinitely because the fluentd RPC endpoint should
+# come up eventually and won't terminate without a signal from outside (barring errors)
+echo '['$(date)']' 'waiting for RPC endpoint to become available'
+until netstat -tln | grep "$RPC_ADDRESS" >/dev/null
+do
+  [ -z "$DEBUG" ] && echo '['$(date)']' 'RPC endpoint not available, waiting'
+  sleep 1
+done
+
+echo '['$(date)']' 'waiting for fluentd to exit' # i.e. stop listening on the RPC address
+while netstat -tln | grep "$RPC_ADDRESS" >/dev/null
+do
+  [ -z "$DEBUG" ] && echo '['$(date)']' 'RPC endpoint still listening'
+
+  if [ "$(find $BUFFER_PATH -iname '*.buffer' -or -iname '*.buffer.meta' | wc -l)" = 0 ]
+  then
+    echo '['$(date)']' 'no buffers left, terminating workers:' "$(curl http://$RPC_ADDRESS/api/processes.killWorkers 2>/dev/null)"
```

Comment:
I wouldn't hide the error output if it's not necessary
```suggestion
    echo '['$(date)']' 'no buffers left, terminating workers:' "$(curl http://$RPC_ADDRESS/api/processes.killWorkers)"
```